example/spec/parallel/discoverGranules/DiscoverGranulesDuplicateHandlingErrorSpec.js

```diff
@@ -0,0 +1,177 @@
+      const sourcePath = `${prefix}/tmp/${randomId('test-')}`;
+
+      // Create the collection
+      collection = await createCollection(
+        prefix,
+        {
+          duplicateHandling: 'error',
+          provider_path: `${sourcePath}/`
+        }
+      );
+
+      // Create the S3 provider
+      provider = await createProvider(prefix, { host: sourceBucket });
+
+      // Stage the existing granule file to S3
+      existingGranuleId = randomId('existing-granule-');
+      existingGranuleKey = `${sourcePath}/${existingGranuleId}.txt`;
+      await s3PutObject({
+        Bucket: sourceBucket,
+        Key: existingGranuleKey,
+        Body: 'asdf'
+      });
+
+      // Ingest the existing granule
+      ingestGranuleRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'IngestGranule',
+          collection: pick(collection, ['name', 'version']),
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-'),
+            granules: [
+              {
+                granuleId: existingGranuleId,
+                dataType: collection.name,
+                version: collection.version,
+                files: [
+                  {
+                    name: `${existingGranuleId}.txt`,
+                    path: sourcePath
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      );
+
+      // Find the "IngestGranule" execution ARN
+      const ingestGranuleExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === ingestGranuleRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+
+      // Wait for the "IngestGranule" execution to be completed
+      await getCompletedExecution({ prefix, arn: ingestGranuleExecutionArn });
+
+      // Wait for the existing granule to be fully ingested
+      await getCompletedGranule({ prefix, granuleId: existingGranuleId });
+
+      // Run DiscoverGranules
+      discoverGranulesRule = await createOneTimeRule(
+        prefix,
+        {
+          workflow: 'DiscoverGranules',
+          collection: {
+            name: collection.name,
+            version: collection.version
+          },
+          provider: provider.id,
+          payload: {
+            testExecutionId: randomId('test-execution-')
+          }
+        }
+      );
+
+      // Find the "DiscoverGranules" execution ARN
+      discoverGranulesExecutionArn = await findExecutionArn(
+        prefix,
+        (execution) =>
+          get(execution, 'originalPayload.testExecutionId') === discoverGranulesRule.payload.testExecutionId,
+        { timeout: 15 }
+      );
+    } catch (err) {
+      beforeAllFailed = true;
+      console.error(err);
+      throw err;
+    }
+  });
+
+  afterAll(async () => {
+    await deleteS3Object(sourceBucket, existingGranuleKey).catch(noop);
```

Comment:
I can refactor it to use `pAll`, which has an option to not fail on the first exception. I'll just need to be careful because there _is_ an order to some of these operations. (You can't delete a collection with an associated rule, for example.)